### ISSUE TITLE
generatedLanguage is invalid when config comes from jsonConfigurationFile

### ISSUE THREAD

**Issue by jxnu-liguobin** [kobylynskyi/graphql-java-codegen#572]
## Issue Description

*Describe your bug in detail.*

When I use JSON as the configuration(generate kotlin), the language can be generated successfully, but it does use Java type, as follows:
```
    val first: Integer,
```
This is a bug.

If I add a separate language configuration in `build.gradle`, it will not appear, as follows:
```
compileKotlin.dependsOn "graphqlCodegenKotlinService"
task graphqlCodegenKotlinService(type: GraphQLCodegenGradleTask) {
    graphqlSchemas.includePattern = "schema\\.graphqls"
    outputDir = new File("src/main/graphql")
    jsonConfigurationFile = "src/main/resources/codegen-config.json"
    generatedLanguage = GeneratedLanguage.KOTLIN // it will be invalid in json, add here
}
```

## Steps to Reproduce
*List in detail the exact steps to reproduce the unexpected behavior.*

## Expected Result
*Explain in detail what behavior you expected to happen.*

## Actual Result
*Explain in detail what behavior actually happened.*

## Your Environment and Setup

* **graphql-java-codegen version**: *X.X.X*
* **Build tool**: *E.g.: Maven/Gradle/SBT*
* **Mapping Config**: *E.g.:* 

```xml
{
  "generateClient": true,
  "generateApis": true,
  "generatedLanguage": "KOTLIN",
  "generateBuilder": false,
  "generateImmutableModels": true,
  "apiPackageName": "io.github.graphql.resolver",
  "modelPackageName": "io.github.graphql.model",
  "generateModelOpenClasses": true,
  "modelNameSuffix": "TO"
}
```

**Comment by jxnu-liguobin** [comment]
@kobylynskyi K, The language judgment of all plug-ins has problems, ignoring the language judgment in JSON.

```java
    private GraphQLCodegen instantiateCodegen(MappingConfig mappingConfig) throws IOException {
        switch (generatedLanguage) {
            case JAVA:
                return new JavaGraphQLCodegen(getSchemas(), graphqlQueryIntrospectionResultPath, outputDir, mappingConfig, buildJsonSupplier(jsonConfigurationFile));
            case SCALA:
                return new ScalaGraphQLCodegen(getSchemas(), graphqlQueryIntrospectionResultPath, outputDir, mappingConfig, buildJsonSupplier(jsonConfigurationFile));
            case KOTLIN:
                return new KotlinGraphQLCodegen(getSchemas(), graphqlQueryIntrospectionResultPath, outputDir, mappingConfig, buildJsonSupplier(jsonConfigurationFile));
            default:
                throw new LanguageNotSupportedException(generatedLanguage);
        }
    }
```

**Comment by kobylynskyi** [comment]
@jxnu-liguobin appreciate your effort for fixing this. Will be released soon as part of 4.1.5
